### ISSUE TITLE
Detect duplicacy when one number selected from contact and another typed manually

### ISSUE THREAD

**Issue by AkankshaBodhankar** [systers/PCSA-Android#158]
One is selected from contacts 
Another one is typed manually
only +91 is additional to the first one

![screenshot_2016-03-14-13-58-53_com peacecorps pcsa](https://cloud.githubusercontent.com/assets/14356938/13738822/74663234-e9ed-11e5-8275-8b22f2d1f16f.png)

**Comment by vihanga-liyanage** [comment]
Apart from detecting the duplicacy what is the expectation hear? Should we delete one number after prompting the user, and if so, which one should we delete?

**Comment by Buddhiprabha** [comment]
@AkankshaBodhankar how do you plan to solve the issue?

**Comment by vihanga-liyanage** [comment]
I would suggest a simple algorithm to track the similarity of the six number as a starter. I'm sure it can be minimized using some conditions as well. As an example, if the number contains a +, then we can get that number, slice away the first three characters and compare with other five numbers.

**Comment by vihanga-liyanage** [comment]
Well follow inputs are also accepted in the current system. This occurs if the contact on the phone was entered without country codes, as it shows in the second image.

![capture](https://cloud.githubusercontent.com/assets/9441136/13810381/920d80cc-eb95-11e5-80e5-ceac485c0a6c.PNG)

![capture2](https://cloud.githubusercontent.com/assets/9441136/13810432/d0cc0f54-eb95-11e5-8bbb-67ede1731743.PNG)

**Comment by AkankshaBodhankar** [comment]
@Buddhiprabha  I was planning to use substring() method for implementation
I think implementing this would be better once validating numbers is done
I have opened an issue for validating numbers as well ISSUE #84 
Once validation is done it will be easier to implement. As, the length of the numbers will be fixed and also we would know exactly which substring to compare with ...like substring(3rd index ,last) in +91'9985397810' will be compared to 9985397810

**Comment by shruthinayak** [comment]
Hello, 
Great work on the app :+1: 
I'm a newbie and wanted to make some contributions. Hence looked into this issue. If I may suggest, here are some things that can be incorporated.  
**Set error flags to EditText instead of showing toast. **
![image](https://cloud.githubusercontent.com/assets/1544918/13831281/7715be82-eba0-11e5-8499-0f6329f9faa0.png)

This will help us achieve a couple things. 
1. Let the user know what he exactly needs to change. 
    To do this, we can maintain a Set of already entered phone numbers. In the "OnActivityResult", before setting the EditText to the selected phone number, one can put a check to see if the Set already contains the phone number, if yes then set the error flag with appropriate message.
`phoneInput.setText(selectedNumber);
                        if (phSet.contains(selectedNumber)) {
                            phoneInput.setError(getString((R.string.duplicate_number_errormessage)));
                            phoneInput.requestFocus();
                        }`
2. Lets say that one of the phone numbers is invalid for ex.(565 or 123456), we can set the error flag with different message. 
   This can be done when we get the text from the EditText after clicking the save button.
3. Code wise: 
   This will simplify the code of check_duplicate_number() to ` if(comrade'x'editText.getError()!=null){
              return true;
          }`

Let me know if this change is okay. I can take it up. 
 
Thanks, 
Shruthi.

**Comment by AkankshaBodhankar** [comment]
@shruthinayak : Great idea. Get in touch with the mentors to get an approval :-)

**Comment by shruthinayak** [comment]
@AkankshaBodhankar Thank you. 
@Buddhiprabha Could you please take a look at the suggestion? Let me know what you think about it.

**Comment by Buddhiprabha** [comment]
@shruthinayak Please follow the discussion at https://github.com/systers/PCSA-Android/issues/84

**Comment by sandarumk** [comment]
On hold until validating numbers is finalized.
